File: src/main/java/com/googongill/aditory/service/dto/user/UpdateUserResult.java
package com.googongill.aditory.service.dto.user;

import com.googongill.aditory.domain.User;
import lombok.Builder;
import lombok.Getter;

@Getter
@Builder
public class UpdateUserResult {
    private Long userId;
    private String nickname;
    private String contact;

    public static UpdateUserResult of(User user) {
        return UpdateUserResult.builder()
                .userId(user.getId())
                .nickname(user.getNickname())
                .contact(user.getContact())
                .build();
    }
}

File: src/main/java/com/googongill/aditory/service/dto/user/UpdateUserResult.kt
package com.googongill.aditory.service.dto.user

import com.googongill.aditory.domain.User

data class UpdateUserResult(
    val userId: Long?,
    val nickname: String?,
    val contact: String?
) {

    companion object {
        fun of(user: User): UpdateUserResult {
            return UpdateUserResult(
                userId = user.id,
                nickname = user.nickname,
                contact = user.contact
            )
        }
    }
}
